gobblin-api/src/main/java/org/apache/gobblin/configuration/ConfigurationKeys.java

```diff
@@ -393,6 +393,7 @@
   public static final String SIMPLE_WRITER_DELIMITER = "simple.writer.delimiter";
   public static final String SIMPLE_WRITER_PREPEND_SIZE = "simple.writer.prepend.size";
   public static final String WRITER_ADD_TASK_TIMESTAMP = WRITER_PREFIX + ".addTaskTimestamp";
+  public static final String DATASET_DEFINED_STAGING_DIR_FLAG = "dataset.defined.staging.dir.flag";
```

Comment:
Shall we rename it ? since it is not clear to me what does "dataset.defined" mean here. Also, Let's avoid words like flag since it is usually ambiguous.